File: less Comfortable/pset6/caesar.py
import sys
import cs50

# ensure that two arguments are provided
if len(sys.argv) == 2:
    key = int(sys.argv[1])
    
    # get plain text
    print("plaintext: ", end = "")
    ptext = cs50.get_string()
    
    print("ciphertext: ", end = "")
    
    # iterate over the plain text
    
    for c in ptext:
        if (c.isalpha() == True):
            if (c.isupper == True):
                # convert to ascii
                num1 = ord(c)
                # subtract 65 to get '0'.add key then take mod with 26.
                codnum1 = (num1 - 65 + key) % 26
                # again add 65 to conver to upper alphabet
                code1 = chr(codnum1 + 65)
                print(code1, end = "")
            
            else:
                # convert to ascii
                num2 = ord(c)
                # subtract 97 to get '0'.add key then mod with 26
                codnum2 = (num2 - 97 + key) % 26
                # again add 97 to convert to lower alphabets
                code2 = chr(codnum2 + 97)
                print(code2, end = "")
        
        else:
            # if not an alphabet, print as it is
            print(c, end = "")
    print("")

else:
    print("ERROR")
    exit(1)
            
        

File: less Comfortable/pset2/caesar/caesar.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>

int main(int argc, string argv[])
{
    // ensure only 2 arguments are provided
    if (argc == 2)
    {
        // take mod of provided key with 26
        int key = atoi(argv[1]);

        // get plain text
        printf("plaintext: ");
        string ptext = get_string();

        printf("ciphertext: ");
        
        // iterate over plaintext
        for (int j = 0, length = strlen(ptext); j < length; j++)
        {
            // verify whether alphabet or not
            if (isalpha(ptext[j]))
            {
                if (isupper(ptext[j]))
                {
                    // convert to ascii
                    char letter = ptext[j];
                    int num1 = (int) letter;
                    // subtract 65 to get '0'.add key then take mod with 26.
                    int codnum1 = (num1 - 65 + key) % 26;
                    // again add 65 to conver to upper alphabet
                    char code1 = (char) (codnum1 + 65);
                    printf("%c", code1);
                }
                else
                {
                    // convert to ascii
                    char letter2 = ptext[j];
                    int num2 = (int) letter2;
                    // subtract 97 to get '0'.add key then mod with 26
                    int codnum2 = (num2 - 97 + key) % 26;
                    // again add 97 to convert to lower alphabets
                    char code2 = (char) (codnum2 + 97);
                    printf("%c", code2);
                }
            }
            else
            {
                // if not alphabet, print as it is
                printf("%c", ptext[j]);
            }
        
        }
    
        printf("\n");
    }
    else
    {
        printf("ERROR\n");
        return (1);
    }
}